File: 861-flipping-an-image/flipping-an-image.py
class Solution:
    def flipAndInvertImage(self, image: List[List[int]]) -> List[List[int]]:
        row = [0] * len(image[0])
        for i in range(len(image)):
            image[i].reverse()
            for j in range(len(image)):
                if image[i][j] == 0: image[i][j] = 1
                else: image[i][j] = 0

        return image

File: 861-flipping-an-image/flipping-an-image.cpp
class Solution {
public:
    vector<vector<int>> flipAndInvertImage(vector<vector<int>>& image) {
        for (auto& row : image) {
            int l = 0, r = row.size() - 1;
            while (l <= r) {
                // Swap and invert using XOR operation
                int temp = row[l] ^ 1;
                row[l] = row[r] ^ 1;
                row[r] = temp;

                // Move pointers towards the center
                l++;
                r--;
            }
        }
        return image;
    }
};
